pkg/cable/wireguard/WGdriver.go

```diff
@@ -0,0 +1,365 @@
+				break
+			}
+		}
+		if !found {
+			klog.Errorf("Failed to verify peer configuration")
+		}
+	}
+	w.peers[remoteEndpoint.Spec.ClusterID] = remoteKey
+
+	// Add routes to peer
+	//TODO save old routes for removal
+	var wg netlink.Link
+	if wg, err = netlink.LinkByName(DefaultDeviceName); err != nil {
+		return "", fmt.Errorf("failed to find wireguard link by name: %v", err)
+	}
+	for _, peerNet := range allowedIPs {
+		route := netlink.Route{
+			LinkIndex: wg.Attrs().Index,
+			Dst:       &peerNet,
+		}
+		if err = netlink.RouteAdd(&route); err != nil {
+			return "", fmt.Errorf("failed to add route %s: %v", route.String(), err)
+		}
+	}
+
+	klog.V(log.TRACE).Infof("Successfully connected endpoint peer %+v", peerCfg)
+
+	return ip, nil
+}
+
+func (w *wireguard) DisconnectFromEndpoint(remoteEndpoint types.SubmarinerEndpoint) error {
+	klog.V(log.TRACE).Infof("Removing endpoint %v+", remoteEndpoint)
+
+	if w.localEndpoint.Spec.ClusterID == remoteEndpoint.Spec.ClusterID {
+		klog.V(log.TRACE).Infof("Will not disconnect self")
+		return nil
+	}
+	var err error
+	var found bool
+
+	// public key
+	var key string
+	if key, found = remoteEndpoint.Spec.BackendConfig[PublicKey]; !found {
+		return fmt.Errorf("missing peer public key")
+	}
+	var remoteKey wgtypes.Key
+	if remoteKey, err = wgtypes.ParseKey(key); err != nil {
+		return fmt.Errorf("failed to parse public key %s: %v", key, err)
+	}
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+	var oldKey wgtypes.Key
+	keyMismatch := false
+	if oldKey, found = w.peers[remoteEndpoint.Spec.ClusterID]; !found {
+		keyMismatch = true
+		klog.Warningf("Key mismatch, cluster %s has no key but asked to remove %s", remoteEndpoint.Spec.ClusterID, remoteKey.String())
+	} else if oldKey.String() != remoteKey.String() {
+		keyMismatch = true
+		klog.Warningf("Key mismatch, cluster %s key is %s but asked to remove %s", remoteEndpoint.Spec.ClusterID, oldKey.String(), remoteKey.String())
+	}
+
+	// wg remove
+	klog.V(log.TRACE).Infof("Removing wireguard peer with key %s", remoteKey.String())
+	peerCfg := []wgtypes.PeerConfig{{
+		PublicKey: remoteKey,
+		Remove:    true,
+	}}
+	if err = w.client.ConfigureDevice(DefaultDeviceName, wgtypes.Config{
+		ReplacePeers: false,
+		Peers:        peerCfg,
+	}); err != nil {
+		return fmt.Errorf("failed to remove wireguard peer with key %s: %v", remoteKey.String(), err)
+	}
+	if keyMismatch {
+		klog.Warningf("Key mismatch for peer cluster %s, keeping existing routes", remoteEndpoint.Spec.ClusterID)
+		return nil
+	}
+	delete(w.peers, remoteEndpoint.Spec.ClusterID)
+
+	// del routes
+	var wg netlink.Link
+	if wg, err = netlink.LinkByName(DefaultDeviceName); err != nil {
+		return fmt.Errorf("failed to find wireguard device by name: %v", err)
+	}
+	var cidr *net.IPNet
+	for _, sn := range remoteEndpoint.Spec.Subnets {
+		if _, cidr, err = net.ParseCIDR(sn); err != nil {
+			return fmt.Errorf("failed to parse subnet %s: %v", sn, err)
```

Comment:
Perhaps just log the error and continue? Either way we shouldn't get a bad CIDR...

Reply:
Done, ignoring malformed subnets. Might create peers with no allowed IPs.